Add stabilization to NMNIST dataset

marcorax opened the pull request:
Hey Gregor! I have been using Tonic and I added a feature for the N-MNIST dataset.

I added an option to stabilize the saccades of the nmnist dataset, as done here https://www.garrickorchard.com/datasets/n-mnist.
I basically ported Garrick matlab function to Python.

This can was useful to me for comparing my SNN on nminst vs an ANN on standard mnist, since it makes NMNIST and MNIST more similar (both centered).

biphasic commented on tonic/datasets/nmnist.py:
```diff
@@ -28,6 +27,7 @@ class NMNIST(Dataset):
         train (bool): If True, uses training subset, otherwise testing subset.
         first_saccade_only (bool): If True, only work with events of the first of three saccades.
                                    Results in about a third of the events overall.
+        stabilize (bool): If True, it stabilize egomotion of the saccades, centering the digit.
```
stabilizeS

biphasic commented on tonic/datasets/nmnist.py:
```diff
@@ -127,3 +131,34 @@ def _check_exists(self) -> bool:
+    """
+    Stabilize digits, code ported from https://www.garrickorchard.com/datasets/n-mnist
+    Returns:
+        stabiliazed events, removing the egomotion caused by saccades.
```
typo in stabilized.

biphasic reviewed (commented):
cool thank you. Could you please use the [Black formatter](https://black.readthedocs.io/en/stable/) on the file you changed? Then I can merge

marcorax commented:
Ok I fixed the formatting and the two errors, 
I also tried to print the digits and found that by porting Garrick code the digits are not centered. 
I added a manual offset to fix that and upper boundaries to check that after translation no events end up with coordinates higher than 33.
This is the result of printing a digit 2 with three frames (at every saccade):
![Screenshot from 2023-08-10 11-25-09](https://github.com/neuromorphs/tonic/assets/32542419/6cc988d1-a8e6-4c12-9eb8-b5d1a05dabd2)

biphasic commented:
Great, merging now. For future commits, please describe the change in the commit message more clearly, so when other people go through the commit history, they can understand what's going! Thank you
